### ISSUE TITLE
[Bug]: Button V9 Teams theme: High contrast text color is white, should be black to meet accessibility requirements

### ISSUE THREAD

**Issue by jurokapsiar** [microsoft/fluentui#22464]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
v9
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://6002298f95a00c00213f4d55-qpafvorpoj.chromatic.com/?path=/docs/components-button--default

## Actual Behavior
color is white

## Expected Behavior
color should be black

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by jurokapsiar** [comment]
Design spec:
[https://www.figma.com/file/Nj9EBBvOZmS11zKNJfilVR/Button?node-id=2360%3A431](https://www.figma.com/file/Nj9EBBvOZmS11zKNJfilVR/Button?node-id=2360:431)
